### ISSUE TITLE
Localizable location?

### ISSUE THREAD

**Issue by jspitz** [plk/biblatex#1390]
Many of the (larger) cities where publishers locate have different or differently spelled names in different languages (e.g., Wien -- Vienna -- Vienne -- Beć, ...; Brüssel -- Brussels -- Bruxelles, ...). In bibliographic entries, the publisher location normally should be given in the main language of the document.

So I wonder whether `\DeclareListFormat{location}` should not use
```tex
\ifbibstring{#1}{\bibstring{#1}}{#1\isdot}%
```
(actually just like patent location). Biblatex could include bibkeys for the most common publisher locations, but even if it didn't, it would help to maintain an own list. Currently I am working around this with `@string`s which change depending on the document language.

Another related issue which I am not sure how to deal with concerns the (mainly US?) convention to give state names (Cambridge, MA), the use and form (cf. Cambridge, Mass.) of which seems to depend on style. Maybe this could be approached via data annotation, or `\locstate{<bibkey>}` macros. Anyways, this seems less central than the issue above.

**Comment by plk** [comment]
I would support this - I always wondered about the `location` and localisation - @moewew?

**Comment by jspitz** [comment]
If you consider to provide a selected list of some common locations, from the top of my head with a brief glance at [this Wikipedia list](https://en.wikipedia.org/wiki/List_of_names_of_European_cities_in_different_languages)

- Antwerp
- Athens
- Basle
- Berne
- Beijing
- Belgrade
- Bolzano
- Brescia
- Brussels
- Cape Town
- Cairo
- Cologne
- Copenhagen
- Florence
- Geneva
- Gothenbourg
- The Hague
- Hanover
- Leuwen
- Lisbon
- Lucerne
- London
- Milan
- Moscow
- Munich
- Nijmegen
- Nuremberg
- Rome
- Saint Petersburg
- Strasbourg
- Tokyo
- Trieste
- Venice
- Vienna
- Zurich

Maybe also:

- Frankford (Frankfurt/M.)

**Comment by moewew** [comment]
Hmmm. Thing is, this will probably be a huge burden to maintain and we probably have to add all sorts of cities once people start asking for it.  (At some point we might even run into performance issues if we take it too far.) Do we really want this?

**Comment by plk** [comment]
We could do a short list - any others need to be added by users - the above list isn't too long and I doubt there would be performance issues with that.

**Comment by pauloney** [comment]
My understanding of the objective might not be complete, but I imagine --
at a minimum -- that this rather incomplete list of 35 cities need to be
translated in the more than 40 supported languages in BibLaTeX. In itself
that does not seem like an easy task

Paulo Ney

**Comment by plk** [comment]
I think the idea is more that we put in the code and commented placeholders to allow this to work and then people can send pull requests etc. to update the various language files over time.

**Comment by jspitz** [comment]
> We could do a short list - any others need to be added by users - the above list isn't too long and I doubt there would be performance issues with that.

That's what I was meaning to suggest.

**Comment by jspitz** [comment]
To expand a bit, one possibility would be to use just the English city names as bibstrings. This way, people will just get the English name if no localization exists in their language.

A drawback of this is that there might be cases where the name is localized without people being aware of it (say, they have "Vienna and London" in their entry and suddenly get "Vienne/Londres" or even "Vienne/London". This would not happen with lowercased keys, but then, the would just get the key ("vienna/London"). On the other hand, they only get this if they explicitly entered it.

**Comment by pauloney** [comment]
Would the list be part of the LBX files?

**Comment by jspitz** [comment]
> Would the list be part of the LBX files?

If it is decided that biblatex provides a selection of common locations, then this goes to the `*.lbx` files, as any other bibstrings. As with other bibstrings as well, any individual additions go elsewhere (for instance, a personal style or the preamble).

Let me rephrase that I think the feature would be very useful even if biblatex itself does not provide any of these strings out of the box (although I think that the list of locations with very common publishers is finite)

**Comment by pauloney** [comment]
The list itself does need some work. "Leuven" writes with a "v" in English;
it is way too Nordic, with only one city in the Southern hemisphere...

I am in a camp that believes the language on a biblio entry should be
unique and (in most cases) the language of the bibliography entry source
and NOT the main language of the document -- because one needs to be able
to locate it and that will give you the best chances.

I really like the standardization and verification of correctness, not only
on this field, but also in several other ones, but a minimal translation
would be really necessary to make it useful.

PN

**Comment by jspitz** [comment]
>  I am in a camp that believes the language on a biblio entry should be unique and (in most cases) the language of the bibliography entry source and NOT the main language of the document -- because one needs to be able to locate it and that will give you the best chances. 

We have `autolang` to control for that, no?

**Comment by pauloney** [comment]
Yes, but in the whole there are a few pieces missing -- coverage in LBX
being one of them. It is easy for a single document in English to demand
several LBX files. We are missing several important ones in the world of
publishing like Latin, Chinese and Japanese. The feature is nice but it
will be as useful as the translation coverage.

PN

**Comment by jspitz** [comment]
I understand the concern, but I don't see how this is related to this requested feature particularly. This is a general issue that concerns all bibstrings. In any case, _having_ the possibility to localize location at all will make a crucial difference to people who need it -- also if they have to come up with their own bibstring translations.

**Comment by jspitz** [comment]
Maybe useful for matching: https://www.publishersglobal.com/directory/list-cities/sort/popular

**Comment by pauloney** [comment]
Uau! Much better list and it is linked to the publishers located there,
which in trade the usefulness of the feature a huge lot!

We would have to write a crawler to get all this info out.

**Comment by pauloney** [comment]
The data is straight from the ISBN database, and a bit on the "old" side,
my guess is about 20 years old...but it has been slightly curated -- it
corrects some typos in the ISBN list.

If anyone knows how to obtain the list from the original source (and
up-to-date) that would be a real plus. We do get ISBN ranges from them in a
regular way - for the ISBN checker - so maybe we can get this too.

**Comment by pauloney** [comment]
Sorry for the mangled text, I was on the phone and auto-correct tripped it
... should have been:

Uau! Much better list and it is linked to the publishers located there,
which will increase the usefulness of the feature a huge lot! We can add
the publishers to the localization.

**Comment by moewew** [comment]
@schoeps This function has not been added to `biblatex` yet (neither the `dev` version here on GitHub nor the release on CTAN). You can see the bulk of what would be needed in https://github.com/plk/biblatex/pull/1403. Then you just need to add the required bibstrings.

As I said above and in https://github.com/plk/biblatex/issues/1410#issuecomment-2669431278, for now I'd like to hold off adding this to the `biblatex` core, because I think without a list of cities we would be promising something we can't deliver on and with a list of cities we are likely to run into a maintenance/political discussion nightmare, but I will count your comment as interest in this feature.

**Comment by schoeps** [comment]
> without a list of cities we would be promising something we can't deliver on and with a list of cities we are likely to run into a maintenance/political discussion nightmare, but I will count your comment as interest in this feature.

@moewew Yes, I am interested and I have made a simple implementation for myself (see above). I think that it is not important to have a full list of cities (this point was already made previously) as this feature breaks nothing. If you put "Paris" inside the location field, everything works as expected. One could even consider to NOT put any city list in biblatex such that people who are interested in this feature need to do something. Actually, this was what I tried first, i.e., defining "Paris" as a bibstring resolving to "Paris, France". 

I even thought to use bibstrings like "Paris" (with capital letter) such that the city name still looks good if the bibstring is undefined but this does not work since bibstrings apparently shall be all lowercase.
